storages/backends/gcloud.py

```diff
@@ -249,10 +252,17 @@ def get_created_time(self, name):
         return created if setting('USE_TZ') else timezone.make_naive(created)
 
     def url(self, name):
-        # Preserve the trailing slash after normalizing the path.
+        """
+        Return public url or a signed url for the Blob.
+        This DOES NOT check for existance of Blob - that makes codes too slow
+        for many use cases.
+        """
         name = self._normalize_name(clean_name(name))
-        blob = self._get_blob(self._encode_name(name))
-        return blob.public_url
+        blob = self.bucket.blob(self._encode_name(name))
+        if (self.default_acl == 'publicRead'):
```

Comment:
Are we covering all possibilities in the [possible ACL list](https://cloud.google.com/storage/docs/access-control/lists) with this logic?

If so, let's remove the parens. Otherwise, what changes are necessary?

Reply:
Technically there is publicReadWrite that allows you to create a bucket that anyone can write to - this seems dangerous, but I can add the check.

Reply:
authenticatedRead only works for Google accounts that have been added to the Blob. Since we have no way on knowing if the user is one of those authenticated people - the safe thing is to send the signed url. 
Basically, if you are going to really use Google ACL and user lists, you will have to use the Google API more directly.